xc/xc7/yosys/synth.tcl

```diff
@@ -25,16 +26,20 @@ if { $::env(USE_ROI) == "TRUE" } {
     read_verilog -lib $::env(TECHMAP_PATH)/iobs.v
 
     hierarchy -check -auto-top
-    update_pll_params
 
     # Start flow after library reading
-    synth_xilinx -vpr -flatten -abc9 -nosrl -noclkbuf -nodsp -iopad -nowidelut -run prepare:check
+    synth_xilinx -vpr -flatten -abc9 -nosrl -nodsp -iopad -nowidelut -run prepare:check
 }
+
 if { [info exists ::env(INPUT_XDC_FILE)] && $::env(INPUT_XDC_FILE) != "" } {
   read_xdc -part_json $::env(PART_JSON) $::env(INPUT_XDC_FILE)
   write_fasm -part_json $::env(PART_JSON)  $::env(OUT_FASM_EXTRA)
 }
 
+# Perform clock propagation based on the information from the SDC commands
+propagate_clocks
```

Comment:
I guess propagate_clocks  should go into the INPUT_XDC_FILE if. It doesn't make anything if we don't read the SDC commands and in this flow we do this in `read_xdc`.

Reply:
Sorry, can you rephrase your statement.  I'm not sure I understand what you are saying.

Reply:
The `propagate_clocks` command should be called inside the `if` statement right after `read_xdc`.

Reply:
Done